packages/plugin/webpack/test/WebpackConfig_spec.ts

```diff
@@ -228,6 +228,59 @@ describe('WebpackConfigGenerator', () => {
         filename: 'preload.js',
       });
     });
+    it('generates a production config', async () => {
+      const config = {
+        renderer: {
+          entryPoints: [{
+            name: 'main',
+            preload: {
+              js: 'preloadScript.js',
+            },
+          }],
+        },
+      } as WebpackPluginConfig;
+      const generator = new WebpackConfigGenerator(config, mockProjectDir, true, 3000);
+      const entryPoint = config.renderer.entryPoints[0];
+      const webpackConfig = await generator.getPreloadRendererConfig(
+        entryPoint,
+        entryPoint.preload!,
+      );
+      expect(webpackConfig.target).to.equal('electron-preload');
+      expect(webpackConfig.mode).to.equal('production');
+      expect(webpackConfig.entry).to.deep.equal(['preloadScript.js']);
+      expect(webpackConfig.output).to.deep.equal({
+        path: path.join(mockProjectDir, '.webpack', 'renderer', 'main'),
+        filename: 'preload.js',
+      });
+    });
+    it('generates config overriding custom target with \'electron-preload\'', async () => {
+      const config = {
+        renderer: {
+          config: {
+            target: 'web',
+          },
+          entryPoints: [{
+            name: 'main',
+            preload: {
+              js: 'preloadScript.js',
+            },
+          }],
+        },
+      } as WebpackPluginConfig;
+      const generator = new WebpackConfigGenerator(config, mockProjectDir, true, 3000);
+      const entryPoint = config.renderer.entryPoints[0];
+      const webpackConfig = await generator.getPreloadRendererConfig(
+        entryPoint,
+        entryPoint.preload!,
+      );
+      expect(webpackConfig.target).to.equal('electron-preload');
+      expect(webpackConfig.mode).to.equal('production');
+      expect(webpackConfig.entry).to.deep.equal(['preloadScript.js']);
+      expect(webpackConfig.output).to.deep.equal({
+        path: path.join(mockProjectDir, '.webpack', 'renderer', 'main'),
+        filename: 'preload.js',
+      });
```

Comment:
You only need to test that the override fails.

```suggestion
    it('prevents the preload target from being overridden', async () => {
      const config = {
        renderer: {
          config: {
            target: 'web',
          },
          entryPoints: [{
            name: 'main',
            preload: {
              js: 'preload.js',
            },
          }],
        },
      } as WebpackPluginConfig;
      const generator = new WebpackConfigGenerator(config, mockProjectDir, true, 3000);
      const entryPoint = config.renderer.entryPoints[0];
      const webpackConfig = await generator.getPreloadRendererConfig(
        entryPoint,
        entryPoint.preload!,
      );
      expect(webpackConfig.target).to.equal('electron-preload');
```